### PR TITLE
fix: add JS for icon property in buttons

### LINKED ISSUE #11142: [Bug]: Add JS for icon property in buttons
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

EX: Starting with the button widget, We see that the button color cannot be dynamically changed and the one in the table widget can. This is an Inconsistency and similar situations can be see across  these properties.

We propose we look at this from the other way around and make all properties JS convertible, and then check what are the failing scenarios.

- [ ] What changes do we need to this from the platform POV.

### Steps To Reproduce

1. D&D button widget
2. try setting the button color dynamically
3. see that you do not really have access to JS

### Environment

Production

### Version

Cloud

### PR DESCRIPTION
**Pull request by sanjus-robotic-studio**
## Description

Add JS functionality for icon in buttons, button group, menu button

Fixes #11142

## Type of change

- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

Tested Manually

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by Rhitottam** [comment]
/perf-test ref=rhito/feat-create-run-meta-before-test-starts-compare-lt-st-for-prev-runs

**Comment by Rhitottam** [comment]
/perf-test ref=rhito/feat-create-run-meta-before-test-starts-compare-lt-st-for-prev-runs

**Comment by sanjus-robotic-studio** [comment]
Hey when will this PR be accepted

**Comment by sanjus-robotic-studio** [comment]
Hey @Tooluloope Can you approve the changes

**Comment by sanjus-robotic-studio** [comment]
@somangshu Can you review this too?

File: app/client/src/widgets/BaseInputWidget/widget/index.tsx [hunk 1]
```diff
@@ -342,7 +342,8 @@ class BaseInputWidget<
                 value: "ITALIC",
               },
             ],
-            isBindProperty: false,
+            isJSConvertible: true,
```
Comment (somangshu) [inline app/client/src/widgets/BaseInputWidget/widget/index.tsx:345]: I dont see where its requested to change the label style to have JS convertible option @sanjus-robotic-studio 

cc @dilippitchika kindly confirm this soon

**Review by somangshu — CHANGES_REQUESTED** [review]
LGTM apart from one small change

**Comment by somangshu** [comment]
@sanjus-robotic-studio while I am looking into the above query, can you merge with latest release branch

File: app/client/src/widgets/BaseInputWidget/widget/index.tsx (hunk 1 above)
Comment (sanjus-robotic-studio) [reply app/client/src/widgets/BaseInputWidget/widget/index.tsx:345]: Hey @somangshu, that commit is from the previous issue #17631. But accidentally it got added to this PR is there any way I can remove that commit

Comment (Tooluloope) [reply app/client/src/widgets/BaseInputWidget/widget/index.tsx:345]: From this comment https://github.com/appsmithorg/appsmith/issues/11142#issuecomment-1277496781 it should only be icons.

Comment (Tooluloope) [reply app/client/src/widgets/BaseInputWidget/widget/index.tsx:345]: Screenshot 2022-10-25 at 13 22 12
Although looking at it now, Select family and all other Widgets have JS convertible  for Label Style except Input family

Comment (sanjus-robotic-studio) [reply app/client/src/widgets/BaseInputWidget/widget/index.tsx:345]: @Tooluloope but I have already created a PR and it got merged for that issue. the pr no is https://github.com/appsmithorg/appsmith/pull/17631

Comment (Tooluloope) [reply app/client/src/widgets/BaseInputWidget/widget/index.tsx]: oh, got it.
Thanks for clarifying that.

**Review by Tooluloope — APPROVED** [review]

File: app/client/src/widgets/BaseInputWidget/widget/index.tsx (hunk 1 above)
Comment (somangshu) [reply app/client/src/widgets/BaseInputWidget/widget/index.tsx]: @sanjus-robotic-studio please remove the change from here; remerge release branch; and push a new commit;

we cant remove the old commit as such. Post that we will be good to go

Comment (sanjus-robotic-studio) [reply app/client/src/widgets/BaseInputWidget/widget/index.tsx]: Hey @somangshu, I have done that and now it is removed

**Review by somangshu — APPROVED** [review]
LGTM

**Comment by somangshu** [comment]
@sanjus-robotic-studio thanks for the changes, the tests are running right now; Ill merge this by tomorrow. 🙏

**Comment by sanjus-robotic-studio** [comment]
Sure, Thanks

**Pull request merged**